app/models/grda_warehouse/data_source.rb

```diff
@@ -574,6 +574,33 @@ def hmis_url_for(entity)
     # For any other Enrollment-related record, link to the enrollment page
     url ||= "#{base}/client/#{entity.client&.id}/enrollments/#{entity.enrollment&.id}" if entity.respond_to?(:enrollment) && entity.respond_to?(:client)
 
+    # If we don't have the HMIS driver we probably aren't here, but we need to check for the next section
+    # If we don't have a user, just return the URl (backwards compatibility)
+    return url if user.blank?
+
+    # If we have a user, check for access (on the HMIS side)
+    hmis_entity = if entity.respond_to?(:as_hmis)
+      entity.as_hmis(force_reload: false)
+    else
+      entity
+    end
+    hmis_user = user.related_hmis_user(self)
+
+    known_permissions = {
+      'GrdaWarehouse::Hud::Project' => [:can_view_project],
+      # 'GrdaWarehouse::Hud::Organization' => [:can_view_organization],
+      'GrdaWarehouse::Hud::Client' => [:can_view_clients],
+      'Hmis::Hud::Enrollment' => [:can_view_projects, :can_view_enrollment_details],
+      'Hmis::Hud::CustomAssessment' => [:can_view_projects, :can_view_enrollment_details],
+    }
+
+    perms = known_permissions[entity.class.name]
+    # If we can't determine if we can see this in HMIS, just go ahead and show the link,
+    # HMIS will handle access
+    return url if perms.blank?
+    # If we can see this in HMIS, don't bother linking to it
+    return nil unless hmis_user.permissions_for?(hmis_entity, perms, mode: :all)
```

Comment:
@eanders this branch isn't working for me I wasn't seeing links anywhere even though I have access. I finally tracked down that this method fails silently if passed a `perms` array... it's supposed to be:

```suggestion
    return nil unless hmis_user.permissions_for?(hmis_entity, *perms, mode: :all)
```

With that fix, at least perms are attempting to resolve, but I get an error:

<img width="1023" alt="Screenshot 2024-03-28 at 4 23 23 PM" src="https://github.com/greenriver/hmis-warehouse/assets/5333982/74ef26ac-f2f7-4736-987c-6f9a7aa134bd">


Sorry that it wasn't so straightforward to figure out the access. I don't mind if this goes into release-110.

Reply:
We have that `persisted?` check there as a regression check measure because we had some sort of bug around permissions on unpersisted records.  Maybe you can trick the model into knowing its persisted in `as_hmis` somehow..